examples/plot_bands_functionality.py

```diff
@@ -0,0 +1,107 @@
+# modules including spatial and plot.
+
+import os
+from glob import glob
+import matplotlib.pyplot as plt
+import earthpy as et
+import earthpy.spatial as es
+import earthpy.plot as ep
+
+# Get data for example
+data = et.data.get_data("vignette-landsat")
+
+###############################################################################
+# Import Example Data
+# -------------------
+#
+# To get started, make sure your directory is set. Then, create a stack from all of
+# the Landsat .tif files (one per band).
+
+os.chdir(os.path.join(et.io.HOME, "earth-analytics"))
+
+# Stack the Landsat 8 bands
+# This creates a numpy array with each "layer" representing a single band
+landsat_path = glob(
+    "data/vignette-landsat/LC08_L1TP_034032_20160621_20170221_01_T1_sr_band*_crop.tif"
+)
+landsat_path.sort()
+array_stack, meta_data = es.stack(landsat_path)
+
+###############################################################################
+# Plot All Bands in a Stack
+# --------------------------
+#
+# When `ep.plot_bands()` is handed a three dimensional numpy array, as is created
+# when `es.stack()` is ran, it will plot all layers of the numpy array. To title
+# all of the images individually, a list of titles must be submitted as an argument.
+# The list must contain the same number of strings as there are bands in the stack.
+
+titles = ["Ultra Blue", "Blue", "Green", "Red", "NIR", "SWIR 1", "SWIR 2"]
+ep.plot_bands(array_stack, title=titles)
+plt.show()
+
+##################################################################################
+# Plot One Band in a Stack
+# ------------------------
+#
+# When `ep.plot_bands` is handed a one dimensional numpy array, such as one layer
+# from the return of `es.stack`, it will just plot that band. If a colorbar is
+# not desired, it can be toggled off.
+
+ep.plot_bands(array_stack[4], cbar=False)
+plt.show()
+
+##################################################################################
+# Turn Off Scaling
+# -----------------
+#
+# `ep.plot_bands()` scales the imagery to a 0-255 scale by default. This can make
+# values easier to compare between images that are scaled differently. To turn off
+# scaling, simply set the argument to false, and the original values will be
+# displayed. Below is an example showing an NDVI calculation with the scaling turned
+# off, in order to display it on a scale from -1 to 1 instead of 0 to 255. Additionally,
+# since it is NDVI, a more appropriate colormap was chosen.
+
+NDVI = es.normalized_diff(array_stack[4], array_stack[3])
+ep.plot_bands(NDVI, scale=False, cmap="RdYlGn")
+plt.show()
+
+##################################################################################
+# Change Shape of Plot
+# ---------------------
+#
+# The number of columns used while plotting multiple bands can be changed in order
+# to change the arrangement of the images overall.
+
+ep.plot_bands(array_stack, cols=2)
```

Comment:
it might be nice to show this using 2 vs 3 columns??

Reply:
3 columns is the default, which has been shown earlier. I could still do a plot with it set to three just to show the user if you want!